core/flyout_base.js

```diff
@@ -466,7 +482,22 @@ Blockly.Flyout.prototype.show = function(xmlList) {
       var tagName = xml.tagName.toUpperCase();
       var default_gap = this.horizontalLayout_ ? this.GAP_X : this.GAP_Y;
       if (tagName == 'BLOCK') {
-        var curBlock = Blockly.Xml.domToBlock(xml, this.workspace_);
+
+        // We assume that in a flyout, the same block id/type means
+        // the same output SVG nodes.  Check for a recycled element.
```

Comment:
You're assuming that the same id/type means the same output `Blockly.BlockSvg` object, not just the same SVG nodes.  Please update the comment.